### PR TITLE
Remove receive address from redux

### PR DESCRIPTION
**Pull request by paullinator**
Perf optimization to prevent needing to call getReceiveAddress() for all wallets each time a wallet update occurs. 

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202060614291040

### CODE REVIEW

File: src/components/themed/WalletListSwipeRow.js [hunk 1]
```diff
@@ -70,7 +76,7 @@ function WalletListSwipeRowComponent(props: Props) {
       guiWallet != null &&
       // Some wallets launch an activation screen when selected,
       // so avoid going to the transaction list in that case:
-      (isToken || !getSpecialCurrencyInfo(guiWallet.type).isAccountActivationRequired || guiWallet.receiveAddress.publicAddress)
+      (isToken || !getSpecialCurrencyInfo(edgeWallet.type).isAccountActivationRequired || walletAddress !== '')
```
Comment (swansontec) [inline src/components/themed/WalletListSwipeRow.js:79]: This is probably not the best solution. The real issue is that sometimes `dispatch(selectWallet)` on the line above will launch the wallet activation scene. In that case, we shouldn't launch the transaction list scene too. If the `selectWallet` action could be modified to return a boolean, we could avoid loading the wallet address altogether in this component.

**Review by swansontec — CHANGES_REQUESTED** [review]
Found a few things, as well as one existing bug that isn't even part of this PR.

File: src/components/scenes/RequestScene.js [hunk 2]
```diff
@@ -127,10 +124,20 @@ export class RequestComponent extends React.Component<Props, State> {
 
   async generateEncodedUri() {
     const { edgeWallet, useLegacyAddress, currencyCode } = this.props
+    let legacyAddress = ''
+    let publicAddress = ''
+    if (edgeWallet != null) {
+      const receiveAddress = await edgeWallet.getReceiveAddress()
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/scenes/RequestScene.js:130]: Can't do it here due to scope

File: src/components/themed/WalletListSwipeRow.js (hunk 1 above)
Comment (paullinator) [reply src/components/themed/WalletListSwipeRow.js:79]: This is a good point, but seems out of scope for this change. I don't believe we have a `walletActivated` boolean in the plugin or core. We are for now relying on the wallet having an address to determine activation.

File: src/modules/UI/scenes/Plugins/EdgeProvider.js [hunk 3]
```diff
@@ -140,20 +141,24 @@ export class EdgeProvider extends Bridgeable {
   async getCurrentWalletInfo(): Promise<WalletDetails> {
     const edgeWallet = this._state.core.account.currencyWallets[this._state.ui.wallets.selectedWalletId]
     const currencyCode = this._state.ui.wallets.selectedCurrencyCode
-    let walletName = edgeWallet.name ? edgeWallet.name : ''
+    let walletName = edgeWallet.name ?? ''
     const receiveAddress = await edgeWallet.getReceiveAddress()
     const enabledTokens = await edgeWallet.getEnabledTokens()
+    const contractAddress = edgeWallet.currencyInfo.metaTokens.find(token => token.currencyCode === currencyCode)?.contractAddress
```
Comment (swansontec) [inline src/modules/UI/scenes/Plugins/EdgeProvider.js:147]: This doesn't account for custom tokens, but then again, those don't have images on our system anyhow. This code is fine, and gets the job done.

The `currencyInfo.metaTokens` field is deprecated, though. The information has moved to the `EdgeCurrencyConfig` object, which in this context would be accessed as:

```js
const currencyConfig = this._state.core.account.currencyConfig[edgeWallet.currencyInfo.pluginId]
currencyConfig.builtinTokens // map of tokenId -> EdgeToken
currencyConfig.customTokens // map of tokenId -> EdgeToken
```

I have a task to make it available as `EdgeCurrencyWallet.currencyConfig`, which would be much shorter to type than this crazy-long thing.

**Review by swansontec — APPROVED** [review]

**Pull request merged**